pkg/registry/cachesize/cachesize.go

```diff
@@ -56,40 +58,27 @@ const (
 	Secrets                    Resource = "secrets"
 	ServiceAccounts            Resource = "serviceaccounts"
 	Services                   Resource = "services"
+	StorageClasses             Resource = "storageclasses"
+
+	// Default value of watch cache size for a resource if not specified.
+	defaultWatchCacheSize = 100
 )
 
+// TODO: This shouldn't be a global variable.
 var watchCacheSizes map[Resource]int
 
 func init() {
 	watchCacheSizes = make(map[Resource]int)
-	watchCacheSizes[CertificateSigningRequests] = 1000
-	watchCacheSizes[ClusterRoles] = 100
-	watchCacheSizes[ClusterRoleBindings] = 100
-	watchCacheSizes[Controllers] = 100
-	watchCacheSizes[Daemonsets] = 100
-	watchCacheSizes[Deployments] = 100
-	watchCacheSizes[Endpoints] = 1000
-	watchCacheSizes[HorizontalPodAutoscalers] = 100
-	watchCacheSizes[Ingress] = 100
-	watchCacheSizes[PetSet] = 100
-	watchCacheSizes[PodDisruptionBudget] = 100
-	watchCacheSizes[Jobs] = 100
-	watchCacheSizes[LimitRanges] = 100
-	watchCacheSizes[Namespaces] = 100
-	watchCacheSizes[NetworkPolicys] = 100
-	watchCacheSizes[Nodes] = 1000
-	watchCacheSizes[PersistentVolumes] = 100
-	watchCacheSizes[PersistentVolumeClaims] = 100
-	watchCacheSizes[Pods] = 1000
-	watchCacheSizes[PodTemplates] = 100
-	watchCacheSizes[Replicasets] = 100
-	watchCacheSizes[ResourceQuotas] = 100
-	watchCacheSizes[ScheduledJobs] = 100
-	watchCacheSizes[Roles] = 100
-	watchCacheSizes[RoleBindings] = 100
-	watchCacheSizes[Secrets] = 100
-	watchCacheSizes[ServiceAccounts] = 100
-	watchCacheSizes[Services] = 100
+}
+
+func InitializeWatchCacheSizes(clusterSize int) {
```

Comment:
@fabioy - when I asked this question to @roberthbailey he said that when you are resizing masters, you will recompute all the parameters that are passed via salt/gci/etc.; it that's the case, it would be enough for us.
Or is it not the case?